docs/pages/how_to_guides/how_add_feature.md

```diff
@@ -123,24 +125,38 @@ class Feature : FeaturePreprocessType<IDType, NNZType, ValueType, FeatureType> {
 	// ...
 	Feature(float alpha, float beta){
 		// ...
-		this->RegisterFunction({kCSRFormat}, FeatureCSR);
-		this->RegisterFunction({kCOOFormat}, FeatureCOO);
+		this->RegisterFunction({format::CSR<IDType, NNZType, ValueType>::get_id_static()}, FeatureCSR);
+		this->RegisterFunction({format::COO<IDType, NNZType, ValueType>::get_id_static()}, FeatureCOO);
 		// ...
 	// ...
 };
 ```
 
 ### 6. Add explicit template instantiations
 
-For the compiled version of the library, you must add explicit instantiations of your class depending on the data types you are going to use for `IDType`, `NNZType`, `ValueType`, and `FeatureType`.
+The functions we have defined so far have been created in header files. This means that they will be compiled as they become needed by the user's code, and not at library build-time. However, sparsebase supports a compiled version in which classes are pre-compiled using certain data types that the user selects. To add your class to the list of pre-compilable classes, you must do the following:
 
-```cpp
-template class Feature<unsigned int, unsigned int, float, double>;
+1. Move all the implementations from the header file (`src/sparsebase/preprocess/preprocess.h`) to the implementation file (`src/sparsebase/preprocess/preprocess.cc`).
+2. Add your class to the list of classes that will be explicitly instantiated by the python script `src/generate_explicit_instantiations.py`.
+
+Step two is much simpler than it sounds. To the file `src/class_instantiation_list.json`, add a single entry containing the `Feature` class definition and the name of the file to which the instantiations are to be printed. In the case of `Feature`, add the following entry to the aformentioned file:
+
+```json
+{
+  "template": "class Feature<$id_type, $nnz_type, $value_type, $feature_type>",
```

Comment:
The last parameter should be `$float_type`.

Reply:
But the feature types does not have to be a float.

Reply:
We only have $id_type, $nnz_type, $value_type, and $float_type as tempalated variable lists provided by CMake, so we need to only use those in the template JSON.